Userland/Applications/Terminal/main.cpp

```diff
@@ -473,5 +473,6 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
     int result = app->exec();
     dbgln("Exiting terminal, updating utmp");
     utmp_update(ptsname, 0, false);
+    TRY(Core::System::kill(shell_pid, SIGTERM));
```

Comment:
Do we really care about any error at this point?

Also, doing SIGTERM will cause a race between Terminal closing and Shell shutting down normally. In order to avoid a race and also make sure that Shell really definitely for sure is killed (instead of "doing something" in the background), we should use SIGKILL:
```suggestion
    auto maybe_error = Core::System::kill(shell_pid, SIGKILL);
    if (maybe_error.is_error())
        dbgln("Could not kill shell: {}", maybe_error.release_error());
```

Reply:
Please give the poor shell some time to die on its own before doing this then, otherwise current history entries won't be written to disk.